lib/tessel/deployment/rust.js

```diff
@@ -33,32 +37,129 @@ var exportables = {
         program
       };
     },
-    shell: () => {
+    shell: (opts) => {
       return tags.stripIndent `
         #!/bin/sh
-        exec /app/remote-script/rust_executable
+        exec /app/remote-script/${opts.resolvedEntryPoint}
       `;
     },
   },
   lists: lists,
 };
 
-exportables.preBundle = function() {
-  return Promise.resolve();
+// The Rust prebundle step just updated the resolvedEntryPoint
+// property with the Rust binary name
+exportables.preBundle = function(opts) {
+  return new Promise((resolve, reject) => {
+    // Get details of the project
+    var details = exportables.meta.checkConfiguration(opts.target);
```

Comment:
This is where I get my failure, even if the TOML is copied.

`opts.target` is set to `"/Users/cflewis/Downloads/hello/src"`, when `checkConfiguration` is expecting three arguments. When L31 is triggered, the `path` package barfs as `program` is undefined. Question is why `opts` is only one rather than three things, and that I did not have time to work out before I had to stop.